spec/cordova/platform/remove.spec.js

```diff
@@ -0,0 +1,160 @@
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var path = require('path');
+var fs = require('fs');
+var Q = require('q');
+var events = require('cordova-common').events;
+var rewire = require('rewire');
+var platform_remove = rewire('../../../src/cordova/platform/remove');
+var platform_metadata = require('../../../src/cordova/platform_metadata');
+var cordova_util = require('../../../src/cordova/util');
+var promiseutil = require('../../../src/util/promise-util');
+var fail;
+
+describe('cordova/platform/remove', function () {
+    var projectRoot = '/some/path';
+    var cfg_parser_mock = function () {};
+    var cfg_parser_revert_mock;
+    var hooks_mock;
+    var package_json_mock;
+    package_json_mock = jasmine.createSpyObj('package json mock', ['cordova', 'dependencies']);
+    package_json_mock.dependencies = {};
+    package_json_mock.cordova = {};
+
+    var hooksRunnerRevert;
+
+    beforeEach(function () {
+        hooks_mock = jasmine.createSpyObj('hooksRunner mock', ['fire']);
+        hooks_mock.fire.and.returnValue(Q());
+        hooksRunnerRevert = platform_remove.__set__('HooksRunner', function () {});
+        cfg_parser_mock.prototype = jasmine.createSpyObj('config parser mock', ['write', 'removeEngine']);
+        cfg_parser_revert_mock = platform_remove.__set__('ConfigParser', cfg_parser_mock);
+        spyOn(fs, 'existsSync').and.returnValue(false);
+        spyOn(fs, 'writeFileSync');
+        spyOn(cordova_util, 'removePlatformPluginsJson');
+        spyOn(events, 'emit');
+        spyOn(cordova_util, 'requireNoCache').and.returnValue({});
+    });
+    afterEach(function () {
+        cfg_parser_revert_mock();
+        hooksRunnerRevert();
+    });
+    describe('error/warning conditions', function () {
+        it('should require specifying at least one platform', function (done) {
+            platform_remove('remove', hooks_mock).then(function () {
+                fail('remove success handler unexpectedly invoked');
+            }).fail(function (e) {
+                expect(e.message).toContain('No platform(s) specified.');
+            }).done(done);
+        });
+    });
+    describe('happy path (success conditions)', function () {
+        it('should fire the before_platform_* hook', function () {
```

Comment:
for some reason, when i run this test I get `Parsing false failed`. But the test passes